addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/MainTVServerService.java

```diff
@@ -176,7 +176,7 @@ public void onServiceSubscribed(String service, boolean succeeded) {
     public void onValueReceived(String variable, String value, String service) {
 
         String oldValue = stateMap.get(variable);
-        if (value.equals(oldValue)) {
+        if (oldValue.equals(value)) {
```

Comment:
I found it, the reasoning was different, the problem might be bigger :-)

If we revert it, it will throw an NPE if the value = null and all if-s contained in the case become unreachable. So I don't know whether it's likely to be null, but at least it was incorrect as it was, what do you think is the proper solution.

All calls come from a same structure as:

```
for (String variable : result.keySet()) {
    onValueReceived(variable, result.get(variable), serviceId);
}
```

So null seems unlikely to me, maybe we should just remove the null checks in the cases of the switch.